File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.io.*;
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> hm = new HashMap<>();
        for (String p : participant) {
            hm.put(p, hm.getOrDefault(p, 0) + 1);
        }
        
        for (String c : completion) {
            int cnt = hm.get(c);
            if (cnt == 1){
                hm.remove(c);
            } else {
                hm.put(c, cnt - 1);
            }
        }
        
        String s = "";
        for (String ss : hm.keySet()) {
            s += ss;
        }
        
        return s;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int> m;
    
    for (int i = 0; i < participant.size(); i++) {
        m[participant[i]]++;
    }
    
    for (int i = 0; i < completion.size(); i++) {
        m[completion[i]]--;
    }
    
    for (auto person : m) {
        if (person.second == 1) {
            return person.first;
        }
    }
    
    return 0;
}